Bug: "file" provisioner only uses first entry from Sources if processing directories

maxim-lobanov opened the pull request:
### Related bug https://github.com/hashicorp/packer/issues/4917
File provisioner supports `sources` property to copy a few entries. It works perfect for files but works incorrectly for directories. Provisioner stops copying after the first directory due to incorrect statement.

**Repro steps:**
```
{
  "type": "file",
  "sources": [
    "SetupDocker",
    "SetupComponentHost"
  ],
  "destination": "C:/Setup"
}
```

[return](https://github.com/hashicorp/packer/blob/v1.6.1/provisioner/file/provisioner.go#L181) statement finishes function execution on the first folder in the list. Correct behavior is just finishing the current iteration of loop and proceed with next one.
Fixing of this bug will allow to reduce and simplify templates significantly.

### Changes
- Replace `return` with `continue` and add error handling.
- Add tests to validate copying of multiple files and multiple directories (multiple directories test failed without my changes)

P.S. It is my first experience with Go, please let me know if something is incorrect with syntax / code.

### Update
Also fixed one more bug with overriding destination path when copying multiple files to folder:
```
{
  "type": "file",
  "sources": [
    "file_1.txt",
    "file_2.txt",
  ],
  "destination": "/something/"
}
```
![image](https://user-images.githubusercontent.com/16715858/89617133-528cb780-d892-11ea-8f81-6290f71058e5.png)

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/packer?pullRequest=9725)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/packer?pullRequest=9725)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/packer?pullRequest=9725).</sub>

maxim-lobanov commented:
> Hi, thanks for this PR! This is a great start, but needs just one more tweak that I can see in order to prevent issues.
@SwampDragons 
Good catch. I have faced with the same issue recently but didn't realize root cause.
Actually it reproduces even for files:
```
{
  "type": "file",
  "sources": [
    "file_1.txt",
    "file_2.txt",
  ],
  "destination": "/something/"
}
```
If `/something/` doesn't exist before this command, it will produce:
![image](https://user-images.githubusercontent.com/16715858/89613412-d6db3c80-d88a-11ea-89fc-e32b5f5e8a47.png)

Fixed using your proposal and added one more test to cover it.
